spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/CrossCol.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+import scala.util.hashing.MurmurHash3
+
+/**
+ * CrossCol operation preforms crosses of categorical features.
+ *
+ * The transformation can be thought of as Hash(cartesian product of features) % hashBucketSize)
+ *
+ * The input Table contains more than or equal to 2 Tensor[String].
+ * Each Tensor[String] represents a categorical feature column.
+ * Each row in Tensor[String] represents the string of value,
+ * which supports single-value and multi-value joined by strDelimiter.
+ *
+ * As for the SparseTensor, it should be transformed to
+ * Tensor[String] before feeding into the Operation
+ *
+ *
+ * For example, if the two input tensors with size=3 are:
+ *  "A,D", "B", "A,C"
+ *  "1", "2", "3,4"
+ *
+ *  the output tensor should be an 2D 3 x hashBucketSize SparseTensor:
```

Comment:
Do we need to update the ```2D 3 x hashBucketSize``` to ```2D 3 x maxLength```?

Reply:
@yiheng my mistake and the update has well done.